File: app/src/main/java/uk/lobsterdoodle/edinburghwolves/app/di/UiComponent.java
package uk.lobsterdoodle.edinburghwolves.app.di;

import javax.inject.Singleton;

import dagger.Component;
import uk.lobsterdoodle.edinburghwolves.app.roster.RosterListFragment;
import uk.lobsterdoodle.edinburghwolves.app.base.App;
import uk.lobsterdoodle.edinburghwolves.core.presenter.RosterListFragmentPresenter;

@Singleton
@Component(modules = { UiModule.class })
public interface UiComponent {

    void inject(App app);

    void inject(RosterListFragment rosterListFragment);

    RosterListFragmentPresenter provideRosterListFragmentPresenter();
}

File: app/src/main/kotlin/uk/lobsterdoodle/edinburghwolves/di/UiComponent.kt
package uk.lobsterdoodle.edinburghwolves.di

import javax.inject.Singleton

import dagger.Component
import uk.lobsterdoodle.edinburghwolves.OverviewFragment
import uk.lobsterdoodle.edinburghwolves.roster.RosterListFragment
import uk.lobsterdoodle.edinburghwolves.app.base.App
import uk.lobsterdoodle.edinburghwolves.app.di.UiModule
import uk.lobsterdoodle.edinburghwolves.core.presenter.RosterListFragmentPresenter
import uk.lobsterdoodle.edinburghwolves.home.FixturesFragment

@Singleton
@Component(modules = arrayOf(UiModule::class))
interface UiComponent {

    fun inject(app: App)

    fun inject(rosterListFragment: RosterListFragment)

    fun inject(overviewFragment: OverviewFragment)

    fun inject(fixturesFragment: FixturesFragment)

    fun provideRosterListFragmentPresenter(): RosterListFragmentPresenter
}
